### PR TITLE
Add natvis for Utf8String and Utf16String

### PR DESCRIPTION
**Pull request by xezon**
natvis

### CODE REVIEW

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -326,6 +326,15 @@ set(GAMEENGINE_WB_SRC
     editor/editmain.cpp
 )
 
+
+set(GAMEENGINE_NATVIS "")
+
+if (MSVC_IDE)
+    list(APPEND GAMEENGINE_NATVIS
+        natvis/thyme.natvis
```
Comment (xezon) [inline src/CMakeLists.txt]: Maybe put file in other folder? I would have put it in root/tools/natvis, but then I would have needed to use ../tools/natvis here.

File: src/CMakeLists.txt [hunk 2]
```diff
@@ -418,11 +427,11 @@ endif()
 
 # Static library for standalone builds and linking tools.
 if(STANDALONE OR BUILD_TOOLS)
-    add_library(thyme_lib STATIC ${GAMEENGINE_SRC} ${GAMEENGINE_HEADERS} game/crashhandler.cpp)
+    add_library(thyme_lib STATIC ${GAMEENGINE_SRC} ${GAMEENGINE_HEADERS} ${GAMEENGINE_NATVIS} game/crashhandler.cpp)
```
Comment (xezon) [inline src/CMakeLists.txt]: Maybe this is wrong place. Natvis file must be added to one project only. If it is added to 2, Visual Studio will actually complain. But `thyme_lib` is conditional with `STANDALONE` and `BUILD_TOOLS`. Where to better put?

File: src/CMakeLists.txt (hunk 1 above)
Comment (OmniBlade) [reply src/CMakeLists.txt]: I would suggest doing this at the top level CMake and have the xml in a new folder for none source resources, something like `resources/msvc/thyme.natvis`, then have it added to the targets after `add_subdirectory(src)` where you can check which targets exist and add it to the most appropriate, something liek the following:

```CMake
if(MSVC)
    if(TARGET thyme_lib)
        target_sources(thyme_lib PRIVATE resources/msvc/thyme.natvis)
    elseif(TARGET thyme_dll)
        target_sources(thyme_dll PRIVATE resources/msvc/thyme.natvis)
    endif()
endif()
```

And just add checks for as many targets as can potentially be built alone?

Comment (xezon) [reply src/CMakeLists.txt]: I opted for similar but different approach: I just created a new INTERFACE library for MSVC_IDE that contains all natvis files.

**Review by OmniBlade — APPROVED** [review]
Looks good to me.

**Pull request merged**